core/kazoo_attachments/src/kz_att_s3.erl

```diff
@@ -78,11 +80,21 @@ get_file_path('undefined', DbName, DocId, AName) ->
 get_file_path(Path, DbName, DocId, AName) ->
     kz_util:to_list(list_to_binary([Path, "/", DbName, "/", DocId, "_", AName])).
 
--spec get_s3_values(ne_binary()) -> {ne_binary(), ne_binary(), ne_binary(), ne_binary(), api_binary()}.
-get_s3_values(S3) ->
+-spec get_s3_values(ne_binary(), kz_data:connection()) ->
+                           {ne_binary(), ne_binary(), ne_binary(), ne_binary(), api_binary()}.
+get_s3_values(S3, 'undefined') ->
     case binary_to_term(base64:decode(S3)) of
         {Key, Secret, Bucket, Path} ->
             {Key, Secret, ?AMAZON_S3_HOST, Bucket, Path};
         {_Key, _Secret, _Host, _Bucket, _Path} = V2 ->
             V2
+    end;
+get_s3_values(S3, HandlerProps) ->
+    try get_map_values(HandlerProps) of
+        {Bucket, Key, Secret, Path, Host} ->
+            {Key, Secret, Host, Bucket, Path}
```

Comment:
@lazedo i have changed the values, let me know if the var naming is correct or should i change it something else :)